packages/asset-swapper/src/utils/quote_simulation.ts

```diff
@@ -148,12 +145,12 @@ export function fillQuoteOrders(
         if (remainingInput.lte(0)) {
             break;
         }
+        result.gas += getTotalGasUsedByFills(fo.order.fills, gasSchedule);
```

Comment:
Why was this calculation moved here? Isn't this wrong? We might not need to hit all the fills/sources.

Reply:
oops, you're right